### DISCUSSION TITLE
Lumerical's 2.5D FDTD in MEEP?

### DISCUSSION THREAD

**Discussion by satadrudas** [NanoComp/meep#2563] [Q&A]
I had been working around inverse designing of integrated photonic devices. The 2D optimizations are pretty quick however, the 3D optimizations are very slow (I don't have access to a computing clusters). 

As the title suggests, I was wondering if here are any tools or techniques in which we can do something like Lumerical's 2.5D FDTD to speed up the simulation of 3D structures .

As mentioned in the Lumerical webpage: "The method involves reducing the 3D problem into an effective 2D problem by converting the vertical waveguide structure into effective dispersive materials that simultaneously account for material and waveguide dispersion" 

Lumerical 2.5D link: https://www.lumerical.com/learn/whitepapers/lumericals-2-5d-fdtd-propagation-method/

**Comment by smartalecH** [comment]
I don't have a ton of faith in these methods wrt topology optimization due to the complicated structure that evolves... but it's worth a try. 

@joamatab you mentioned someone had 2.5D kinda working within meep, right?

**Reply by smartalecH** [reply]
Note that even if someone has a 2.5D variant working, the adjoint code isn't setup to differentiate that operator (so your gradients will be wrong).

**Reply by satadrudas** [reply]
Thank you so much. Guess I'll have to run and compare the results.

**Comment by stevengj** [comment]
I think all these 2.5d methods do is calculate an "effective index" from the dispersion relation of a 3d planar waveguide, and then use that to replace the actual index in a 2d simulation.  You can easily do that yourself by calculating the dispersion relation in MPB.   Sometimes people include other heuristics too.

Quantitatively, these methods are pretty useless for complex geometries (like those produced by topology optimization).  *Qualitatively*, we almost always start with a 2d simulation just to get the math right and to identify a reasonable parameter regime, but I don't think the 2.5D heuristic is necessary for this.

**Reply by satadrudas** [reply]
Thanks for the input. Did a few readings and understood that 2.5d wouldn't be of much help in complex geometry like you mentioned.

**Comment by satadrudas** [comment]
Came across a paper relevant to this discussion:

[Inverse-designed low-index-contrast structures on a silicon photonics platform for vector–matrix multiplication](https://www.nature.com/articles/s41566-024-01394-2)
